### PR TITLE
[MG5] Added GammaGammaToQQ cards

### PR DESCRIPTION
**Pull request by gkrintir**
Dear GEN convs,

some MG5 cards (in v265) following the

`[1908.05180](https://arxiv.org/abs/1908.05180)`

preprint for gg->qq' generation.

Regards

### CODE REVIEW

File: bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTobb/GammaGammaTobb_run_card.dat [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+#**************************************
+# Parameter below of the systematics study
+#  will be used by SysCalc (if installed)
+#**************************************
+#
+0.5 1 2 = sys_scalefact  # factorization/renormalization scale factor
+0.5 1 2 = sys_alpsfact  # lpha_s emission scale factors
+auto = sys_matchscale # variation of merging scale
+# PDF sets and number of members (0 or none for all members).
+NNPDF23_lo_as_0130_qed = sys_pdf # list of pdf sets. (errorset not valid for syscalc)
```
Comment (agrohsje) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTobb/GammaGammaTobb_run_card.dat:295]: Please remove.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTocc/GammaGammaTocc_run_card.dat [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+#*********************************************************************
+# Beam polarization from -100 (left-handed) to 100 (right-handed)    *
+#*********************************************************************
+     0.0     = polbeam1 ! beam polarization for beam 1
+     0.0     = polbeam2 ! beam polarization for beam 2
+
+#*********************************************************************
+# PDF CHOICE: this automatically fixes also alpha_s and its evol.    *
+#*********************************************************************
+  nn23lo1	= pdlabel ! PDF set
```
Comment (agrohsje) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTocc/GammaGammaTocc_run_card.dat:51]: Is this intended ?

**Comment by agrohsje** [comment]
I am confused about the flavor scheme and the merging strategy. Can you comment on cc versus bb what concerns additional jets?

File: bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTobb/GammaGammaTobb_run_card.dat (hunk 1 above)
Comment (gkrintir) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTobb/GammaGammaTobb_run_card.dat:295]: OK sure

File: bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTocc/GammaGammaTocc_run_card.dat (hunk 2 above)
Comment (gkrintir) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTocc/GammaGammaTocc_run_card.dat:51]: Right, this is what PDF used in the provided LHE files from the authors (for the gg->tautau process)

**Comment by gkrintir** [comment]
Thx @agrohsje for the comments.

For the flavor scheme and the merging could you elaborate a bit more whether you see smth wrong with the setup and/or concerns on possible double counting in the cc and bb samples?

File: bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTocc/GammaGammaTocc_run_card.dat (hunk 2 above)
Comment (agrohsje) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTocc/GammaGammaTocc_run_card.dat:51]: How will you use this? pLHE + GENSIM? Which tune are you going to use?

**Comment by agrohsje** [comment]
All your quarks up to the b are massless. Did you check the impact from making the b massive? Are you sensitive e.g. to differences between charm and other light jets?

File: bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTocc/GammaGammaTocc_run_card.dat (hunk 2 above)
Comment (gkrintir) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTocc/GammaGammaTocc_run_card.dat:51]: I was thinking wmLHEGS; CP5 as in HIN-HINPbPbAutumn18pLHE-00013, etc

**Comment by gkrintir** [comment]
Thanks, Alex, the comment on the b massive quarks is a very good one; I think indeed we can generate 2 samples as massless and massive (now that I see in 1908.05180 the generation was done with massive b's). I am adding the relevant bb cards then.

Authors in 1908.05180 found no impact from either ss, uu or dd.

**Comment by gkrintir** [comment]
Hi Alex, now the bb is split in 4 and 5f schemes.

Regards

File: bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTobb_5f_ckm/GammaGammaTobb_5f_ckm_run_card.dat [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+# Apply pt/E/eta/dr/mij/kt_durham cuts on decay products or not
+# (note that etmiss/ptll/ptheavy/ht/sorted cuts always apply)
+#*********************************************************************
+  False	= cut_decays ! Cut decay products 
+#*********************************************************************
+# Standard Cuts                                                      *
+#*********************************************************************
+# Minimum and maximum pt's (for max, -1 means no cut)                *
+#*********************************************************************
+  20.0	= ptj ! minimum pt for the jets
```
Comment (agrohsje) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTobb_5f_ckm/GammaGammaTobb_5f_ckm_run_card.dat:117]: In the 4FS you don't cut on the b pT. As this is 5FS and maxjetflavor 5 you will treat the b as jet, so if I am not mistaken, this cut will apply. Why not making consistent cuts of 10 GeV or so? Assuming this is low enough.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTocc/GammaGammaTocc_run_card.dat [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+# Apply pt/E/eta/dr/mij/kt_durham cuts on decay products or not
+# (note that etmiss/ptll/ptheavy/ht/sorted cuts always apply)
+#*********************************************************************
+  False	= cut_decays ! Cut decay products 
+#*********************************************************************
+# Standard Cuts                                                      *
+#*********************************************************************
+# Minimum and maximum pt's (for max, -1 means no cut)                *
+#*********************************************************************
+  20.0	= ptj ! minimum pt for the jets
```
Comment (agrohsje) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTocc/GammaGammaTocc_run_card.dat:117]: As above. That will be different in the different samples.

**Comment by agrohsje** [comment]
Thanks for all the changes. I removed the 2 obsolete files. I have a final question on cuts. Otherwise this looks good to me. Did you start testing?

File: bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTobb_5f_ckm/GammaGammaTobb_5f_ckm_run_card.dat (hunk 3 above)
Comment (gkrintir) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTobb_5f_ckm/GammaGammaTobb_5f_ckm_run_card.dat]: 10 GeV now put to all jet flavors

File: bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTocc/GammaGammaTocc_run_card.dat (hunk 4 above)
Comment (gkrintir) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/GammaGammaToQQ/GammaGammaTocc/GammaGammaTocc_run_card.dat]: 10 GeV now put to all jet flavors

**Comment by gkrintir** [comment]
thx @agrohsje and sorry for the late followup

the requests have been set, I'll notify the HIN contacts to re-validate with the updated gridpacks:

https://cms-pdmv.cern.ch/mcm/requests?member_of_campaign=HINPbPbAutumn18wmLHEGS&page=0&shown=127

Regards

**Comment by agrohsje** [comment]
Thanks @gkrintir: Looks good to me. @GurpreetSinghChahal should have a final check and merge. 
One comment, did you modify the PDF in the systematics command so that NNPDF 2.3 variations are included? Otherwise you will have no corresponding PDF variations.

**Comment by gkrintir** [comment]
thanks @agrohsje I didn't touch the syst; I could add indeed the NNPDF 2.3 variations although these samples will be used as part of bkg in our case so we don't need per se but those weights could be useful indeed in general. I am having a look.

**Comment by agrohsje** [comment]
You need this in the run_card
True = store_rwgt_info ! Store info for reweighting in LHE file
and then you can also afterwards modify the systematics line in runcmsgrid.sh.

**Comment by agrohsje** [comment]
Hi @gkrintir . These are the cards as we submitted the request these days? Let me know and I will merge. :-)

**Comment by gkrintir** [comment]
Hi @agrohsje, that's correct, many thanks.

**Pull request merged**